### PR TITLE
Pass raw string to yaml.BytesUnmarshaler

### PR DESCRIPTION
**Pull request by kyoh86**
fix: #75

### CODE REVIEW

File: decode.go [hunk 1]
```diff
@@ -336,7 +336,12 @@ func (d *Decoder) decodeValue(dst reflect.Value, src ast.Node) error {
 	}
 	valueType := dst.Type()
 	if unmarshaler, ok := dst.Addr().Interface().(BytesUnmarshaler); ok {
-		b := fmt.Sprintf("%v", src)
+		var b string
+		if scalar, isScalar := src.(ast.ScalarNode); isScalar {
+			b = scalar.GetValue().(string)
```
Comment (goccy) [inline decode.go]: `scalar.GetValue()` will return not only `string` type, but also `int64` or `uint64` and so on .
I think better to use `fmt.Sprint(scalar.GetValue())` to convert to string value .
( also, would you please add test case for integer type ? )

File: decode_unmarshalable_string_test.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package yaml
```
Comment (goccy) [inline decode_unmarshalable_string_test.go]: Would you please add test case in `decode_test.go` ?

File: decode.go (hunk 1 above)
Comment (kyoh86) [reply decode.go]: OK, I did it.

File: decode_unmarshalable_string_test.go (hunk 2 above)
Comment (kyoh86) [reply decode_unmarshalable_string_test.go]: done.

**Comment by kyoh86** [comment]
@goccy I corrected the points that were pointed out.

File: decode.go [hunk 3]
```diff
@@ -1000,3 +1005,135 @@ func (d *Decoder) Decode(v interface{}) error {
 	}
 	return nil
 }
+
+type unmarshalableStringValue string
```
Comment (goccy) [inline decode.go]: You should add test code to `decode_test.go` **NOT** `decode.go`

Comment (kyoh86) [reply decode.go]: oh

**Comment by kyoh86** [comment]
fixed.

File: decode_test.go [hunk 4]
```diff
@@ -1355,3 +1357,135 @@ complecated: string
...
+		t.Parallel()
+		var container unmarshalableStringContainer
+		if err := yaml.Unmarshal([]byte(`value: 'aaa'`), &container); err != nil {
+			t.Fatalf("failed to unmarshal %v", err)
+		}
+		if container.V != "aaa" {
+			t.Fatalf("expected \"aaa\", but %q is set", container.V)
+		}
+	})
+	t.Run("(json) empty string", func(t *testing.T) {
```
Comment (goccy) [inline decode_test.go]: Would you please remove test case for `json.Unmarshal` ( also, remove `encoding/json` package )

Comment (kyoh86) [reply decode_test.go]: I thought that is good way to present that yaml.Unmarshal and json.Unmarshal provide same behavior.
I'll remove it.

**Comment by kyoh86** [comment]
fixed.

**Comment by goccy** [comment]
Thanks !!

LGTM

**Pull request merged**